osf/models/schema_response.py

```diff
@@ -192,3 +264,158 @@ def _update_response(self, current_block, latest_response):
             revised_block.save()
             self.response_blocks.remove(current_block)
             self.response_blocks.add(revised_block)
+
+    def delete(self, *args, **kwargs):
+        if self.state is not ApprovalStates.IN_PROGRESS:
+            raise SchemaResponseStateError(
+                'Cannot delete SchemaResponse with id [{self.id}]. In order to delete, '
+                'state must be "in_progress", but is "{self.reviews_state}" instead.'
+            )
+        super().delete(*args, **kwargs)
+
+# *** Callbcks in support of ApprovalsMachine ***
+
+    def _validate_trigger(self, event_data):
+        '''Any additional validation to confirm that a trigger is being used correctly.
+
+        For SchemaResponses, use this to confirm that the provided user has permission to
+        execute the trigger, including enforcing correct usage of the internal "accept" shortcut.
+        '''
+        user = event_data.kwargs.get('user')
+        trigger = event_data.event.name
+
+        # The only valid case for not providing a user is the internal accept shortcut
```

Comment:
Minor: is there any precedent for, or would it make sense to add any flags to `event_data` in cases like these? e.g. `'auto': True` or `'internal': True`.